### PR TITLE
First send screen

### PR DESCRIPTION
**Pull request by x-goose-x**
Add first screen:

- Can add recipient from clipboard.
- Beginnings of QR support (need to resolve permissions issues)
- Need to add choose contact

status:ready

### CODE REVIEW

File: src/status_im/ui/screens/events.cljs [hunk 1]
```diff
@@ -277,3 +278,10 @@
   :update-geolocation
   (fn [db [_ geolocation]]
     (assoc db :geolocation geolocation)))
+
```
Comment (yenda) [inline src/status_im/ui/screens/events.cljs]: did you rebase ? this shouldn't be part of the PR

**Review by yenda — CHANGES_REQUESTED** [review]

File: src/status_im/ui/screens/events.cljs (hunk 1 above)
Comment (x-goose-x) [reply src/status_im/ui/screens/events.cljs]: Right, I rebased from develop. I'll remove this.

File: src/status_im/components/toolbar_new/view.cljs [hunk 2]
```diff
@@ -47,7 +47,13 @@
    [rn/view {:style tst/toolbar-title-container}
     [rn/text {:style (merge tst/toolbar-title-text title-style)
               :font  :toolbar-title}
-     title]]))
+     title]])
+  ([title-style title subtitle-style subtitle]
```
Comment (jeluard) [inline src/status_im/components/toolbar_new/view.cljs]: Can you refactor so that there is no duplication but a single hiccup representation with conditionals?

File: src/status_im/ui/screens/wallet/main/views.cljs [hunk 3]
```diff
@@ -67,10 +67,12 @@
     [react/view {:style wallet.styles/value-variation}
      [react/text {:style wallet.styles/value-variation-title}
       (i18n/label :t/wallet-total-value)]
-     [change-display change]]
+     [react/view {:style (if (pos? change) wallet.styles/today-variation-container-positive wallet.styles/today-variation-container-negative)}
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/main/views.cljs]: Why the change here?

File: src/status_im/ui/screens/wallet/send/styles.cljs [hunk 4]
```diff
@@ -31,4 +32,67 @@
...
+   :padding-bottom  32
+   :justify-content :center})
+
+(def choose-recipient-label
+  {:color :white})
+
+(def recipient-buttons
+  {:flex-direction   :column
+   :margin-left      28
+   :margin-right     28
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/send/styles.cljs]: You can use `margin-horizontal`

File: src/status_im/ui/screens/wallet/send/styles.cljs [hunk 5]
```diff
@@ -31,4 +32,67 @@
...
+   :margin-top       10
+   :border-radius    5
+   :background-color st/color-blue5})
+
+(def recipient-icon {:margin-right 20})
+
+(def recipient-button
+  {:flex-direction  :row
+   :justify-content :space-between
+   :margin-top      10
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/send/styles.cljs]: You can use `margin-vertical`

File: src/status_im/ui/screens/wallet/send/styles.cljs [hunk 6]
```diff
@@ -31,4 +32,67 @@
...
+  {:color     :white
+   :font-size 17})
+
+(defn recipient-touchable [divider?]
+  (cond-> {:border-color st/color-gray-transparent-medium-light}
+    divider? (assoc :border-bottom-width 1)))
+
+(def qr-container
+  {:flex         1
+   :margin-left  32
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/send/styles.cljs]: You can use `margin-horizontal`

File: src/status_im/ui/screens/wallet/send/subs.cljs [hunk 7]
```diff
@@ -0,0 +1,6 @@
+(ns status-im.ui.screens.wallet.send.subs
+  (:require [re-frame.core :as re-frame]))
+
+(re-frame/reg-sub :camera-dimensions
+                  (fn [db]
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/send/subs.cljs]: Indentation

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 8]
```diff
@@ -1,17 +1,73 @@
...
-  [toolbar/toolbar2 {:style cst/toolbar}
-   [toolbar/nav-button (act/back-white act/default-handler)]
-   [toolbar/content-title {:color :white} "Send Transaction"]])
+  [toolbar/toolbar2 {:style send.styles/toolbar}
+   [toolbar/nav-button (act/close-white act/default-handler)]
+   [toolbar/content-title {:color :white} (i18n/label :t/wallet-send-transaction)
+    {:color styles/color-light-blue} (i18n/label :t/wallet-send-step-one)]
+   [toolbar/actions [{:icon      :icons/ok
+                      :icon-opts {:color :white}
+                      :handler   #()}]]])
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/send/views.cljs]: Maybe display some popup?

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 9]
```diff
@@ -1,17 +1,73 @@
...
 (defview send-transaction []
-  []
-  [rn/view {:style cst/wallet-container}
-   [toolbar-view]
-   [rn/text "Nothing here yet!"]])
+  (letsubs [camera-initialized? (r/atom false)
+            camera-dimensions [:camera-dimensions]]
+           [rn/view {:style send.styles/wallet-container}
+     [toolbar-view]
+     [rn/view
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/send/views.cljs]: Can you rename `rn` to `react`?

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 10]
```diff
@@ -1,17 +1,73 @@
...
+      {:style send.styles/choose-recipient-container}
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/send/views.cljs]: We usually define `opts` on the same line as the hiccup element type.

**Review by jeluard — CHANGES_REQUESTED** [review]

**Comment by flexsurfer** [comment]
could you rebase and squash commits?

File: src/status_im/ui/screens/wallet/main/views.cljs (hunk 3 above)
Comment (x-goose-x) [reply src/status_im/ui/screens/wallet/main/views.cljs]: This probably came from `develop`, will revert it.

File: src/status_im/ui/screens/wallet/send/styles.cljs [hunk 11]
```diff
@@ -4,16 +4,16 @@
...
    :background-color styles/color-blue4})
 
-(defstyle toolbar
-  {:ios     {:background-color styles/color-blue4}
-   :android {:background-color styles/color-blue5
-             :elevation        0}})
+(def toolbar
+  {:background-color styles/color-blue5
+   :elevation        0
+   :padding-bottom 10})
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/send/styles.cljs]: indentation

File: src/status_im/ui/screens/wallet/send/events.cljs [hunk 12]
```diff
@@ -0,0 +1,8 @@
+(ns status-im.ui.screens.wallet.send.events
+  (:require [re-frame.core :as re-frame]
+            [status-im.ui.screens.wallet.db :as wallet.db]))
+
+(re-frame/reg-event-db
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/send/events.cljs:5]: imo, we can use `(dispatch [:set-in [:wallet :send :recipient] recipient ])` it's readable right from view, it's just discussion i don't say we need to do it here

File: src/status_im/ui/screens/wallet/send/styles.cljs [hunk 13]
```diff
@@ -30,4 +30,71 @@
...
+
+(defn recipient-touchable [divider?]
+  (cond-> {:border-color styles/color-gray-transparent-medium-light}
+    divider? (assoc :border-bottom-width 1)))
+
+(def qr-container
+  {:flex              1
+   :margin-horizontal 32})
+
+(def preview {:flex            1
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/send/styles.cljs]: indentation

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 14]
```diff
@@ -1,17 +1,78 @@
...
-  [toolbar/toolbar2 {:style cst/toolbar}
-   [toolbar/nav-button (act/back-white act/default-handler)]
-   [toolbar/content-title {:color :white} "Send Transaction"]])
+  [toolbar/toolbar2 {:style send.styles/toolbar}
+   [toolbar/nav-button (act/close-white act/default-handler)]
+   [toolbar/content-title {:color :white} (i18n/label :t/wallet-send-transaction)
+    {:color styles/color-light-blue} (i18n/label :t/wallet-send-step-one)]
+   [toolbar/actions [{:icon      :icons/flash-active
+                      :icon-opts {:color :white}
+                      :handler show-not-implemented!}]]])
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/send/views.cljs]: indentation

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 15]
```diff
@@ -1,17 +1,78 @@
...
+(defn recipient-buttons []
+  [react/view {:style send.styles/recipient-buttons}
+   [react/touchable-highlight {:style (send.styles/recipient-touchable true)}
+    [react/view {:style send.styles/recipient-button}
+     [react/text {:style send.styles/recipient-button-text}
+      (i18n/label :t/wallet-choose-from-contacts)]
+     [vector-icons/icon :icons/contacts {:color           :white
+                                         :container-style send.styles/recipient-icon}]]]
+   [react/touchable-highlight {:style    (send.styles/recipient-touchable true)
+                            :on-press #(react/get-from-clipboard
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/send/views.cljs]: error because indentation

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 16]
```diff
@@ -1,17 +1,78 @@
...
+      (i18n/label :t/wallet-browse-photos)]
+     [vector-icons/icon :icons/browse {:color           :white
+                                       :container-style send.styles/recipient-icon}]]]])
 
 (defview send-transaction []
-  []
-  [rn/view {:style cst/wallet-container}
-   [toolbar-view]
-   [rn/text "Nothing here yet!"]])
+  (letsubs [camera-initialized? (r/atom false)
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/send/views.cljs]: `camera-initialized?` not used

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/ui/screens/wallet/send/styles.cljs (hunk 13 above)
Comment (x-goose-x) [reply src/status_im/ui/screens/wallet/send/styles.cljs]: Isn't this correctly indented?

**Comment by flexsurfer** [comment]
please do not forget to rebase and squash commits after adding new commits, thanks

File: src/status_im/ui/screens/wallet/send/styles.cljs (hunk 13 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/send/styles.cljs]: map should be on the new line, like above
